Bug: range(n) single-argument form doesn't parse

## Description

The `range()` function with a single argument (e.g., `range(5)`) fails to parse. The parser expects two arguments.

## Current Behavior

```
error[E2002]: expected ,, got ) instead
  --> test.ez:1:17
   |
1 | for i in range(5) {
   |                 ^ expected token not found
```

## Expected Behavior

`range(5)` should be equivalent to `range(0, 5)` producing values 0, 1, 2, 3, 4.

## Reproduction

```ez
for i in range(5) {
    println(i)
}
```

## Workaround

Use the two-argument form: `range(0, 5)`

## Possible Fix

Update the range parsing logic to accept 1, 2, or 3 arguments:
- `range(end)` → start=0, end=end, step=1
- `range(start, end)` → start=start, end=end, step=1
- `range(start, end, step)` → start=start, end=end, step=step

--- a/pkg/ast/ast.go
+++ b/pkg/ast/ast.go
@@ -241,11 +241,12 @@ type NewExpression struct {
 func (n *NewExpression) expressionNode()      {}
 func (n *NewExpression) TokenLiteral() string { return n.Token.Literal }
 
-// RangeExpression represents range(start, end)
+// RangeExpression represents range(end), range(start, end), or range(start, end, step)
 type RangeExpression struct {
 	Token Token
-	Start Expression
+	Start Expression // nil for range(end) form, defaults to 0
 	End   Expression
+	Step  Expression // nil for default step of 1
 }
 
 func (r *RangeExpression) expressionNode()      {}

--- a/pkg/interpreter/evaluator.go
+++ b/pkg/interpreter/evaluator.go
@@ -1030,36 +1030,78 @@ func evalForStatement(node *ast.ForStatement, env *Environment) Object {
 				"    for_each item in collection { ... }")
 	}
 
-	startObj := Eval(rangeExpr.Start, env)
-	if isError(startObj) {
-		return startObj
+	// Handle start - defaults to 0 if nil (single-argument form)
+	var start int64 = 0
+	if rangeExpr.Start != nil {
+		startObj := Eval(rangeExpr.Start, env)
+		if isError(startObj) {
+			return startObj
+		}
+		startInt, ok := startObj.(*Integer)
+		if !ok {
+			return newError("range start must be integer")
+		}
+		start = startInt.Value
 	}
+
+	// Handle end
 	endObj := Eval(rangeExpr.End, env)
 	if isError(endObj) {
 		return endObj
 	}
-
-	start, ok := startObj.(*Integer)
-	if !ok {
-		return newError("range start must be integer")
-	}
-	end, ok := endObj.(*Integer)
+	endInt, ok := endObj.(*Integer)
 	if !ok {
 		return newError("range end must be integer")
+	}
+	end := endInt.Value
+
+	// Handle step - defaults to 1 if nil
+	var step int64 = 1
+	if rangeExpr.Step != nil {
+		stepObj := Eval(rangeExpr.Step, env)
+		if isError(stepObj) {
+			return stepObj
+		}
+		stepInt, ok := stepObj.(*Integer)
+		if !ok {
+			return newError("range step must be integer")
+		}
+		step = stepInt.Value
+		if step == 0 {
+			return newError("range step cannot be zero")
+		}
 	}
 
 	loopEnv := NewEnclosedEnvironment(env)
 
-	for i := start.Value; i < end.Value; i++ { // Exclusive end
-		loopEnv.Set(node.Variable.Value, &Integer{Value: i}, true) // loop vars are mutable
+	// Handle positive and negative steps
+	if step > 0 {
+		for i := start; i < end; i += step {
+			loopEnv.Set(node.Variable.Value, &Integer{Value: i}, true)
 
-		result := Eval(node.Body, loopEnv)
-		if result != nil {
-			if result.Type() == RETURN_VALUE_OBJ || result.Type() == ERROR_OBJ {
-				return result
+			result := Eval(node.Body, loopEnv)
+			if result != nil {
+				if result.Type() == RETURN_VALUE_OBJ || result.Type() == ERROR_OBJ {
+					return result
+				}
+				if result.Type() == BREAK_OBJ {
+					break
+				}
 			}
-			if result.Type() == BREAK_OBJ {
-				break
+		}
+	} else {
+		// Negative step: count down
+		for i := start; i > end; i += step {
+			loopEnv.Set(node.Variable.Value, &Integer{Value: i}, true)
+
+			result := Eval(node.Body, loopEnv)
+			if result != nil {
+				if result.Type() == RETURN_VALUE_OBJ || result.Type() == ERROR_OBJ {
+					return result
+				}
+				if result.Type() == BREAK_OBJ {
+					break
+				}
 			}
 		}
 	}

--- a/pkg/parser/parser.go
+++ b/pkg/parser/parser.go
@@ -2411,19 +2411,51 @@ func (p *Parser) parseRangeExpression() Expression {
 	}
 
 	p.nextToken()
-	exp.Start = p.parseExpression(LOWEST)
+	firstArg := p.parseExpression(LOWEST)
 
+	// Check if this is the single-argument form: range(end)
+	if p.peekTokenMatches(RPAREN) {
+		p.nextToken() // consume RPAREN
+		// range(end) -> start=nil (defaults to 0), end=firstArg, step=nil (defaults to 1)
+		exp.Start = nil
+		exp.End = firstArg
+		exp.Step = nil
+		return exp
+	}
+
+	// Two or three argument form
 	if !p.expectPeek(COMMA) {
 		return nil
 	}
 
 	p.nextToken()
-	exp.End = p.parseExpression(LOWEST)
+	secondArg := p.parseExpression(LOWEST)
+
+	// Check if this is the two-argument form: range(start, end)
+	if p.peekTokenMatches(RPAREN) {
+		p.nextToken() // consume RPAREN
+		// range(start, end) -> start=firstArg, end=secondArg, step=nil (defaults to 1)
+		exp.Start = firstArg
+		exp.End = secondArg
+		exp.Step = nil
+		return exp
+	}
+
+	// Three argument form: range(start, end, step)
+	if !p.expectPeek(COMMA) {
+		return nil
+	}
+
+	p.nextToken()
+	thirdArg := p.parseExpression(LOWEST)
 
 	if !p.expectPeek(RPAREN) {
 		return nil
 	}
 
+	exp.Start = firstArg
+	exp.End = secondArg
+	exp.Step = thirdArg
 	return exp
 }
 

--- a/test/range_variants_test.ez
+++ b/test/range_variants_test.ez
@@ -1,0 +1,53 @@
+import @std
+
+do main() {
+    using std
+
+    println("=== Range Variants Tests ===")
+    println("")
+
+    println("Test 1: range(5) - single argument (0 to 4)")
+    print("  ")
+    for i in range(5) {
+        print("${i} ")
+    }
+    println("")
+
+    println("Test 2: range(2, 7) - two arguments (2 to 6)")
+    print("  ")
+    for i in range(2, 7) {
+        print("${i} ")
+    }
+    println("")
+
+    println("Test 3: range(0, 10, 2) - step of 2")
+    print("  ")
+    for i in range(0, 10, 2) {
+        print("${i} ")
+    }
+    println("")
+
+    println("Test 4: range(0, 10, 3) - step of 3")
+    print("  ")
+    for i in range(0, 10, 3) {
+        print("${i} ")
+    }
+    println("")
+
+    println("Test 5: range(10, 0, -2) - negative step (countdown)")
+    print("  ")
+    for i in range(10, 0, -2) {
+        print("${i} ")
+    }
+    println("")
+
+    println("Test 6: range(5, 0, -1) - countdown by 1")
+    print("  ")
+    for i in range(5, 0, -1) {
+        print("${i} ")
+    }
+    println("")
+
+    println("")
+    println("=== All Range Variants Tests Passed! ===")
+}

